src/ReportManager/SchemaFormContext/SchemaForm/fields/Text/index.js

```diff
@@ -0,0 +1,41 @@
+import React, { useContext } from 'react';
+
+import { SchemaFormContext } from '../../SchemaFormContext';
+import { isSchemaFieldRequired } from '../../SchemaFormContext/utils';
+import useFieldDetails from '../../useFieldDetails';
+
+const VALIDATION_ERROR_TYPES = {
+  REQUIRED: 'REQUIRED'
+};
+
+export const TextFieldValidators = {
+  [VALIDATION_ERROR_TYPES.REQUIRED]: (fieldValue, fieldName, schema) => {
+    return isSchemaFieldRequired(schema, fieldName) ? !fieldValue : false;
+  }
+};
+
+const Text = ({ fieldName }) => {
+  const { onFieldChange } = useContext(SchemaFormContext);
+  const textFieldDetails = useFieldDetails(fieldName);
+
+  const handleOnChange = (e) => onFieldChange(fieldName, e.currentTarget.value);
+
+  return <div data-testid={`schema-form-text-field-${fieldName}`}>
+    <label htmlFor={fieldName}>{textFieldDetails.label}</label>
+    <input id={fieldName}
+           value={textFieldDetails.value}
```

Comment:
Another nit: this kind of indentation wastes lots of space. All those empty spaces at the beginning of the line pushes the code further to the right, which then the linter will want to break because it passes the max chars per line limit.